Backend internationalization

mtteo23 opened the pull request:
## TL;DR
changed every hardcoded string (except for the raised errors) for the user to a dictionary in the locales/[lang]/string.json, in order to permit the translation in other languages, partially solving the issue #252 (still there isn't a language selector)

## What
What is affected by this PR?
- [X] API
- [ ] GUI
- [ ] Hardware Integration
- [ ] OS/Deployment
- [ ] Documentation
- [ ] Other (please describe in notes)

## Testing
How was this tested?
- [X] Locally Virtualized
- [ ] Raspberry Pi
- [ ] With Hardware
- [ ] Other (please describe in notes)

## Notes
I'm not sure why the last pull request failed, I didn't change any python configuration

sparklestheunicorn commented:
@mtteo23 Came here to review but this was closed 11 hours ago. Are you making more changes or was there some issue?

sparklestheunicorn commented:
@mtteo23 Also, there's a language selector in the UI but no way yet to send that setting to the backend as far as I know.

mtteo23 commented:
> @mtteo23 Came here to review but this was closed 11 hours ago. Are you making more changes or was there some issue?

Both this and the precedent fail some test that run automatically when the pull request is done, I'm not sure what those test do, it seems to be about the version of some python library.

I planned of trying to resolve them before reopening the pull request (even if they don't seem to be about anything I've changed).

If you prefer to see yourself I'm reopening now the pull request.

mtteo23 commented:
> @mtteo23 Also, there's a language selector in the UI but no way yet to send that setting to the backend as far as I know.

I saw, but I still haven't studied well how the config are handled in this project

mtteo23 commented:
@sparklestheunicorn is it possible to locally run the same test without making a pull request? I'm having trubles identifying the problematic changes and I don't want to clog you with dozens of pull request while I search the error.
(When I try the codes in localhost everything works, so I can't find the problem there)

sparklestheunicorn commented:
> @sparklestheunicorn is it possible to locally run the same test without making a pull request? I'm having trubles identifying the problematic changes and I don't want to clog you with dozens of pull request while I search the error. (When I try the codes in localhost everything works, so I can't find the problem there)

It's not a problem to have a PR open while you're working on something, but we do appreciate you not spamming them.

You can run the back-end tests in a virtual environment using the following commands:

```bash
cd backend
virtualenv -p python3 --system-site-packages env
source env/bin/activate
```

Then in the virtual env:

```bash
(env) $ pip install -r requirements.txt
(env) $ pytest
```

This should run the back-end tests. It looks like the errors on your PRs might be unrelated to the code changes and have more to do with the test runner environment having trouble managing packages. I'll create an issue to investigate. If the tests pass locally, push up the PR and someone from the team can review and try running locally.

sparklestheunicorn commented on backend/microlab/interface.py:
```diff
@@ -29,7 +32,7 @@ def close_to_microlab_queue(self):
                 break
 
-        self._logger.debug('Completed purge of to microlab queue')
+        self._logger.debug('purged-queue')
```
Do we need a t[...] here?

sparklestheunicorn reviewed (approved):
This looks good to me, modulo the potential missing lookup for `'purged-queue'`. Approving so this can be merged once that's updated. Thanks @mtteo23!

mtteo23 commented:
> This looks good to me, modulo the potential missing lookup for `'purged-queue'`. Approving so this can be merged once that's updated. Thanks @mtteo23!

Sorry it took me this long (and for the error in the first place). I fixed the error and removed some unused lines from strings.json. It should now be all fixed

sparklestheunicorn commented:
@mtteo23 Looks great. Thanks!
